root/static/scripts/edit/URLCleanup.js

```diff
@@ -3230,13 +3230,21 @@ const CLEANUPS = {
         'https://twitter.com/',
       );
       url = url.replace(
-        /^(https:\/\/twitter\.com)\/@?([^\/?#]+(?:\/status\/\d+)?)(?:[\/?#].*)?$/,
-        '$1/$2',
+        /^(https:\/\/twitter\.com\/)@?(?:i\/web|([^\/?#]+))?(\/status\/\d+)?(?:[\/?#].*)?$/,
+        function(_, website, username, status) {
+          if (status) {
+            return website + "i/web"+ status;
+          }
+          if (username) {
+            return website + username;
+          }
+          return website;
+        },
       );
       return url;
     },
     validate: function (url, id) {
-      const m = /^https:\/\/twitter\.com\/[^\/?#]+(\/status\/\d+)?$/.exec(url);
+      const m = /^https:\/\/twitter\.com\/(?:[^\/?#]+|(i\/web\/status\/\d+)?)$$/.exec(url);
```

Comment:
Is the double `$$` intentional?
Also, if I'm reading this correctly, then the previous version rejected https://twitter.com/ (the main Twitter page) while this accepts it. Is the `?` after `(i\/web\/status\/\d+)` needed? IMO it makes sense to block linking to just Twitter without a profile or tweet.

Reply:
Sorry, the double `$` was a mistake.
And yes you're right, but I found that though `https://twitter.com/` was blocked, `https://twitter.com` wasn't because it is not matched, should we do something about it?

Reply:
Not sure what @yvanzo thinks but I see no reason not to also block `https://twitter.com` by making the end slash optional on the original matching regex or removing it entirely.